### PR TITLE
Simplify register and bind

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

Syscall registration and binding are currently disjoint operations requiring the callers to explicitly do both for each syscall.

This can lead to mismatch or missing registration/binds and complicates the code. Another side effect is that it is difficult/impossible given the current API for callers to filter syscalls at the registration stage but allow them at the binding stage. One example is the effort to deprecate/disallow syscalls by preventing new deployments of programs that rely on a disallowed syscall while continuing to support existing programs that already rely on them.

#### Summary of Changes

These changes depend and build off the changes here: https://github.com/solana-labs/rbpf/pull/293

Syscalls are registered once, and later the bind operation uses a common context to bind all the previously registered syscalls. Thus, no longer requires an explicit bind for each syscall and without the complications of keeping two identical lists of syscalls for registration and bind.

Fixes #

### CODE REVIEW

**Comment by jeffwashington** [comment]
@jackcmay I rebased this to master to pick up the coverage test fix.

**Comment by jeffwashington** [comment]
I am testing this now on a bare metal mnb validator.

**Comment by jeffwashington** [comment]
appears normal on mnb validator

File: programs/bpf/rust/sysvar/src/lib.rs [hunk 1]
```diff
@@ -110,5 +113,15 @@ pub fn process_instruction(
     sysvar::stake_history::id().log();
     let _ = StakeHistory::from_account_info(&accounts[9]).unwrap();
 
+    // Fees
```
Comment (Lichtso) [inline programs/bpf/rust/sysvar/src/lib.rs:116]: Unintended changes introduced here, possibly while resolving a merge conflict?

Comment (jackcmay) [reply programs/bpf/rust/sysvar/src/lib.rs:116]: this is a new test I added to catch the failure that led to the fees sysvar not working

**Pull request merged**